### PR TITLE
balanced serial iterator

### PR DESCRIPTION
**Pull request by corochann**

### CODE REVIEW

**Comment by corochann** [comment]
I introduced `BalancedSerialIterator`, to train in-balanced data.
This iterator samples each label examples same number of time in one epoch.

Thus, positive (few) examles are sampled several times in one epoch to "balance" with negative (many) examples.

**Comment by corochann** [comment]
Review ready.
If you want to know the behavior of this iterator, please see `test_balanced_serial_iterator.py` or please run tox21 example of the PR https://github.com/pfnet-research/chainer-chemistry/pull/62.

File: chainer_chemistry/iterators/balanced_serial_iterator.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+        indices = []
+        if self.current_pos + num < self.index_length:
+            indices.append(self.current_index_list[
+                           self.current_pos: self.current_pos + num])
+            self.current_pos += num
+        else:
+            indices.append(self.current_index_list[self.current_pos:])
+            num -= (self.index_length - self.current_pos)
+            q, r = divmod(num, self.index_length)
```
Comment (mottodora) [inline chainer_chemistry/iterators/balanced_serial_iterator.py]: Please add comment like 
"In in-balanced data setting, `num` is sometimes more than twice bigger than `self.index_length`"

File: chainer_chemistry/iterators/balanced_serial_iterator.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        indices_list = []
+        for label, index_iterator in self.labels_iterator_dict.items():
+            if label in self.ignore_labels:
+                # Not include index of ignore_labels
+                continue
+            indices_list.append(index_iterator.get_next_indices(
+                self.max_label_count))
+
+        if self._batch_balancing:
+            indices = numpy.array(indices_list).transpose().ravel()
```
Comment (mottodora) [inline chainer_chemistry/iterators/balanced_serial_iterator.py]: Can you add comment in this line?  It takes a bit time to understand.

**Pull request merged**